src/Ocelot/Cache/CacheKeyGenerator.cs

```diff
@@ -1,43 +1,20 @@
-﻿using Ocelot.Configuration;
-using Ocelot.Request.Middleware;
+﻿using Ocelot.Request.Middleware;
 
 namespace Ocelot.Cache
 {
     public class CacheKeyGenerator : ICacheKeyGenerator
     {
-        private const char Delimiter = '-';
-        
-        public async ValueTask<string> GenerateRequestCacheKey(DownstreamRequest downstreamRequest, DownstreamRoute downstreamRoute)
+        public string GenerateRequestCacheKey(DownstreamRequest downstreamRequest)
         {
-            var builder = new StringBuilder()
-                .Append(downstreamRequest.Method)
-                .Append(Delimiter)
-                .Append(downstreamRequest.OriginalString);
-
-            var cacheOptionsHeader = downstreamRoute?.CacheOptions?.Header;
-            if (!string.IsNullOrEmpty(cacheOptionsHeader))
-            {
-                var header = downstreamRequest.Headers
-                    .FirstOrDefault(r => r.Key.Equals(cacheOptionsHeader, StringComparison.OrdinalIgnoreCase))
-                    .Value?.FirstOrDefault();
-
-                if (!string.IsNullOrEmpty(header))
-                {
-                    builder.Append(Delimiter)
-                        .Append(header);
-                }
-            }
-
-            if (!downstreamRequest.HasContent)
+            var downStreamUrlKeyBuilder = new StringBuilder($"{downstreamRequest.Method}-{downstreamRequest.OriginalString}");
+            if (downstreamRequest.Content != null)
             {
-                return MD5Helper.GenerateMd5(builder.ToString());
+                var requestContentString = Task.Run(async () => await downstreamRequest.Content.ReadAsStringAsync()).Result;
```

Comment:
I'm not sure it is a good practice to get result like this

Reply:
No, there is some refactoring needed here but that's some code back in 2019...

Reply:
We will fix the next 2nd round while feature redevelopment.
I guess it is not major issue now.

Reply:
@RaynaldM Not our team, not even the author, but Mr. Ali added this staff in 2019. 😉